src/views/docs-view/index.tsx

```diff
@@ -43,33 +43,76 @@ const DocsView = ({
 	projectName,
 	layoutProps,
 	outlineItems,
+	pageHeading,
 }: DocsViewProps) => {
 	const currentProduct = useCurrentProduct()
 	const { compiledSource, scope } = mdxSource
 	const docsMdxComponents = getDocsMdxComponents(currentProduct.slug)
-	const Layout = layouts[metadata?.layout?.name] ?? DefaultLayout
+
+	/**
+	 * Check if we have a `pageHeading` to render. The `DocsPageHeading` element
+	 * is used on nearly all docs pages, to render a heading element alongside an
+	 * optional version selector. On docs "landing" pages, the heading element
+	 * takes on additional styles (see `LandingHero`).
+	 *
+	 * The one use case where we don't have this is for Packer plugin docs,
+	 * which uses this `DocsView` component directly but uses `docs-view/server`
+	 * indirectly, with some other modifications, such as adding badges above the
+	 * page heading. Packer plugins MDX processing does _not_ remove the `<h1 />`
+	 * from MDX content, so we do not want to render a duplicative
+	 * `DocsPageHeading` `h1` element.
+	 *
+	 * The Packer plugins use case will fade away after Integrations work,
+	 * at which point we can always safely render <DocsPageHeading />.
+	 * Task: https://app.asana.com/0/1202097197789424/1204412156894157/f
```

Comment:
I've tried to be thorough here on documenting that Packer plugins is the exception here.

Feels worth revisiting this as soon as Packer plugins are replaced by integrations, so I've made a follow-up task: 🎟 [[cleanup] Remove docs page heading cruft once /packer/plugins is replaced by integrations](https://app.asana.com/0/1202097197789424/1204412156894157/f)